numpy/lib/format.py

```diff
@@ -317,12 +317,14 @@ def _write_array_header(fp, d, version=None):
     if len(header) >= (256*256) and version == (1, 0):
         raise ValueError("header does not fit inside %s bytes required by the"
                          " 1.0 format" % (256*256))
-    if len(header) < (256*256):
+    elif len(header) < (256*256):
+        if version is None:
+            version = (1, 0)
         header_len_str = struct.pack('<H', len(header))
-        version = (1, 0)
```

Comment:
I think defaulting to (1, 0) is fine, and we still do that, but silently ignoring the user's request for (2, 0) is not okay.